libraries/standard/http/include/private/http_client_internal.h

```diff
@@ -0,0 +1,23 @@
+#ifndef HTTP_CLIENT_INTERNAL_H_
+#define HTTP_CLIENT_INTERNAL_H_
+
+/* Standard Includes. */
+#include <stdint.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "http_client.h"
+
+#define STRLEN_LITERAL( x )    ( ( sizeof( x ) / sizeof( char ) ) - 1 )
+#define HTTP_SUCCEEDED( x )    ( ( x ) != HTTP_SUCCESS )
+#define HTTP_FAILED( x )       ( ( x ) != HTTP_SUCCESS )
+    HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
```

Comment:
I think all internal functions should be static and prototyped+defined in the C file.